### PR TITLE
feat: added `formatValue` param

### LINKED ISSUE #22: FR: with new verison, no way to format "double" value as displayText
My usecase is to show % progress.

Earlier the display would be something like - "20%"

After the recent version, where double values where allowed the display looks like - "20.0%"

I would want to override how the text is shown - I do not want to show decimal places.

I have following solutions in mind:
- add `Widget? display` that can override the "display part completely"
- OR add `String Function(double value) valueToText = _defaultValueToText` - this will be called to "serialize" double value - default could be `(v) => v.toDouble().toString()`

Let me know what is ok for you, will file PR accordingly.

### PR DESCRIPTION
**Pull request by daadu**
Closes #22

### CODE REVIEW

File: lib/flutter_animation_progress_bar.dart [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import 'package:flutter/animation.dart';
 import 'package:flutter/widgets.dart';
 
+String _defaultValueToText(double value) => value.toString();
```
[the remark this answers is not in the document]
Comment (daadu) [reply lib/flutter_animation_progress_bar.dart:4]: also I think `valueToText` is not a good name, other alternatives are `formateValue`, `stringifyValue`.

Let me know your view on it, will modify it accordingly.

[the remark this answers is not in the document]
Comment (ltdangkhoa) [reply lib/flutter_animation_progress_bar.dart:4]: I think ``formatValue`` would be a nice name. Once you commit with new name, I'll merge and publish new version soon

**Comment by daadu** [comment]
@ltdangkhoa when can this change be expected on pub.dev?

**Comment by ltdangkhoa** [comment]
@daadu as my latest test in example, I think toStringAsFixed(2) will display a unnessesary .00 while not in fractional value. I'm going to add this one:
```
String _defaultFormatValue(double value) =>
    value.toStringAsFixed(value.truncateToDouble() == value ? 0 : 2);
```

Do you accept this solution?

**Comment by daadu** [comment]
It is just a matter of preference for "the default" - anyways could be overridden by the end user.

Go ahead if you think that is the right default, I gave my views before.

**Comment by ltdangkhoa** [comment]
I'll add a param in that function for user the right to pick. By default if none set, it will fall back to the shorter one

**Comment by ltdangkhoa** [comment]
> @ltdangkhoa when can this change be expected on pub.dev?

I've released new version on pub.dev, if you found any issue, please raise a new one in this repo.

Many thanks 🙏

**Pull request merged**